File: Sample/.Net/PublicApiDemo/Common/PlexePublicAPIDemo.Common/ApplicationService.cs
using Newtonsoft.Json;
using PlexePublicAPIDemo.Common.Model;
using RestSharp;
using System;
using System.Collections.Generic;
using System.Net;
using System.Text;

namespace PlexePublicAPIDemo.Common
{
    public class ApplicationService
    {
        private readonly string BASE_URI = "https://apidemo.plexe.co/"; // "https://apidemo.plexe.co/";
        private readonly string APPLICATION_URI = "api/Application/";

        public ApplicationResponse GetApplicationById(string key, string applicationId)
        {
            ApplicationResponse applicationResponse = new ApplicationResponse();
            try
            {
                var client = new RestClient(BASE_URI + APPLICATION_URI + "get-application/" + applicationId + "");
                var request = new RestRequest();
                request.Method = Method.Get;
                request.AddHeader("Accept", "application/json");
                request.AddHeader("Authorization", "Bearer " + key + "");
                RestResponse response = client.Execute(request);
                Console.WriteLine(Environment.NewLine + response.Content);
                applicationResponse = JsonConvert.DeserializeObject<ApplicationResponse>(response.Content);
                return applicationResponse;
            }
            catch (Exception ex)
            {
                return null;

            }

        }
        public ApplicationOffer GetApplicationOffer(string key, string applicationId)
        {
            ApplicationOffer applicationOffer = new ApplicationOffer();
            try
            {
                var client = new RestClient(BASE_URI + APPLICATION_URI + "application-offer/" + applicationId + "");
                var request = new RestRequest();
                request.Method = Method.Get;
                request.AddHeader("Accept", "application/json");
                request.AddHeader("Authorization", "Bearer " + key + "");
                RestResponse response = client.Execute(request);
                Console.WriteLine(Environment.NewLine + response.Content);
                applicationOffer = JsonConvert.DeserializeObject<ApplicationOffer>(response.Content);
                return applicationOffer;
            }
            catch (Exception ex)
            {
                return null;

            }

        }
        public ApplicationAndCustomerCreateResponse CreateCustomerAndApplication(string key)
        {
            ApplicationAndCustomerCreateResponse applicationAndCustomerCreateResponse = new ApplicationAndCustomerCreateResponse();
            applicationAndCustomerCreateResponse = null;
            try
            {
                var client = new RestClient(BASE_URI + APPLICATION_URI + "create-customer-and-application");
                var request = new RestRequest();
                request.Method = Method.Post;
                request.AddHeader("Content-Type", "application/json");
                request.AddHeader("Accept", "application/json");
                request.AddHeader("Authorization", "Bearer " + key + "");

                CustomerAndApplication customerAndApplication = new CustomerAndApplication();                
                customerAndApplication.application = new Application()
                {
                    BusinessName = "New Application 24105 Apiuser16092022-1 <TBA>, apiuser16092022-1@plexe.co.au, 8888888888, Sep 16, 2022",
                    ExtraInformation = new ExtraInformation()
                    {
                        Years = "2-5 years",
                        Zipcode = "82772", 
                        Industry = "Rice"
                    },
                    PartnerId = null,
                };
                customerAndApplication.customer = new Customer()
                {
                    Email = "apiuser16092022-1@plexe.co.au",
                    FirstName = "apiuser16092022-1",
                    LastName = "lastname",
                    MobileNumber = "999999999",
                    Password = "DelightWAY12!@"
                };

                var body = JsonConvert.SerializeObject(customerAndApplication);
                request.AddParameter("application/json", body, ParameterType.RequestBody);
                RestResponse response = client.Execute(request);
                Console.WriteLine(response.Content);
                Console.WriteLine(Environment.NewLine + response.Content);
                applicationAndCustomerCreateResponse = JsonConvert.DeserializeObject<ApplicationAndCustomerCreateResponse>(response.Content);
                return applicationAndCustomerCreateResponse;
            }
            catch (Exception ex)
            {
                return null;
            }
        }
        public bool AddOrUpdateBankingData(string key, string applicationId, string bankName)
        {
            ApplicationAndCustomerCreateResponse applicationAndCustomerCreateResponse = new ApplicationAndCustomerCreateResponse();
            applicationAndCustomerCreateResponse = null;
            try
            {
                var client = new RestClient(BASE_URI + APPLICATION_URI + "add-or-update-banking-data/" + applicationId + "?bankName=" + bankName + "");
                var request = new RestRequest();
                request.AddHeader("Content-Type", "application/json");
                request.AddHeader("Accept", "application/json");
                request.AddHeader("Authorization", "Bearer " + key);
                request.Method = Method.Post;

                BankingDataRequest bankingDataRequest = new BankingDataRequest();

                List<BankAccountRequest> bankAccountRequests = new List<BankAccountRequest>();
                List<TransactionRequest> transactionRequests1 = new List<TransactionRequest>();
                List<TransactionRequest> transactionRequests2 = new List<TransactionRequest>();


                transactionRequests1.Add(new TransactionRequest()
                {
                    Amount = 7363,
                    Balance = 345,
                    Date = new DateTime(),
                    Text = "Transaction One",
                    IsCredit = true,
                });
                transactionRequests1.Add(new TransactionRequest()
                {
                    Amount = 4574,
                    Balance = 345,
                    Date = new DateTime(),
                    Text = "Transaction Two",
                    IsCredit = true,
                });
                transactionRequests2.Add(new TransactionRequest()
                {
                    Amount = 7363,
                    Balance = 345,
                    Date = new DateTime(),
                    Text = "Transaction One",
                    IsCredit = true,
                });
                transactionRequests2.Add(new TransactionRequest()
                {
                    Amount = 4574,
                    Balance = 345,
                    Date = new DateTime(),
                    Text = "Transaction Two",
                    IsCredit = true,
                });
                bankAccountRequests.Add(new BankAccountRequest()
                {
                    AvailableBalance = "23333",
                    CurrentBalance = "3455",
                    AccountNumber = "82722892920",
                    Routing = "827282",
                    AccountType = "Current",
                    Bank = "Bank One",
                    Transactions = transactionRequests1,
                    AccountName = "Bank One Test"
                });

                bankAccountRequests.Add(new BankAccountRequest()
                {
                    AvailableBalance = "23333",
                    CurrentBalance = "3455",
                    AccountNumber = "82722892920",
                    Routing = "827282",
                    AccountType = "Current",
                    Bank = "Bank Two",
                    Transactions = transactionRequests2,
                    AccountName = "Bank Two Test"
                });

                bankingDataRequest.BankAccounts = bankAccountRequests;

                var body = JsonConvert.SerializeObject(bankingDataRequest);
                request.AddParameter("application/json", body, ParameterType.RequestBody);
                RestResponse response = client.Execute(request);
                Console.WriteLine(response.Content);
                return true;

            }
            catch (Exception ex)
            {
                return false;
            }
        }
        public bool ProcessApplication(string key, string applicationId)
        {
            try
            {
                var client = new RestClient(BASE_URI + APPLICATION_URI + "process-application/" + applicationId);
                var request = new RestRequest();
                request.Method = Method.Get;
                request.AddHeader("Accept", "application/json");
                request.AddHeader("Authorization", "Bearer " + key + "");
                RestResponse response = client.Execute(request);
                Console.WriteLine(Environment.NewLine + response.Content);
                return true;
            }
            catch (Exception ex)
            {
                return false;
            }
        }
        public string ApplicationStatus(string key, string applicationId)
        {
            try
            {
                var client = new RestClient(BASE_URI + APPLICATION_URI + "application-status/" + applicationId + "");
                var request = new RestRequest();
                request.Method = Method.Get;
                request.AddHeader("Accept", "application/json");
                request.AddHeader("Authorization", "Bearer " + key + "");
                RestResponse response = client.Execute(request);
                Console.WriteLine(Environment.NewLine + response.Content);
                return response.Content;
            }
            catch (Exception ex)
            {
                return null;
            }
        }
    }
}

File: Sample/Php/PublicApiDemo/services/ApplicationService.php
<?php
require_once 'vendor/autoload.php';
use GuzzleHttp\Client;
use GuzzleHttp\Exception\RequestException;
use GuzzleHttp\Psr7\Request;

class ApplicationService
{
    private $BASE_URI = "https://apidemo.plexe.co/";
    private $APPLICATION_URI = "api/Application/";

    function StatusCodeHandling($e)
    {
        if ($e -> getResponse() -> getStatusCode() == '400')
        {
            //$response = json_decode($e -> getResponse()->getBody(true)->getContents());
            //return $response;
            return 'BAD REQUEST';
            //$this->prepare_access_token();
        }
        else if ($e -> getResponse() -> getStatusCode() == '422')
        {
            $response = json_decode($e -> getResponse()->getBody(true)->getContents());
            return $response;
        }
        else if ($e -> getResponse() -> getStatusCode() == '500')
        {
            $response = json_decode($e ->getResponse()->getBody(true)->getContents());
            return $response;
        }
        else if ($e -> getResponse() -> getStatusCode() == '401')
        {
            $response = json_decode($e ->getResponse()->getBody(true)->getContents());
            return $response;
        }
        else if ($e->getResponse()->getStatusCode() == '403')
        {
            $response = json_decode($e->getResponse()->getBody(true)->getContents());
            return $response;
        }
        else
        {
            $response = json_decode($e->getResponse()->getBody(true)->getContents());
            return $response;
        }
    }

    public function GetApplicationById($key, $applicationId)
    {
        try
        {
            $client = new Client();

            $headers = [
                'Accept' => 'application/json',
                'Authorization' => 'Bearer '.$key
            ];

            $request = new Request('GET', $this->BASE_URI.$this->APPLICATION_URI.'get-application/'.$applicationId, $headers);
            
            $res = $client->sendAsync($request)->wait();
            
            $result = $res->getBody();
            $decoded = json_decode($result);
            return $decoded;
        }
        catch (RequestException $e)
        {            
            return 'BAD_REQUEST';
        }
    }       

    public function CreateCustomerAndApplication($key)
    {
        try
        {
            $client = new Client();
            $guid = bin2hex(openssl_random_pseudo_bytes(16));
                    
            $headers = [
                'Content-Type' => 'application/json',
                'Accept' => 'application/json',
                'Authorization' => 'Bearer '.$key
            ];
            $body = '{
                "application": {
                    "businessName": "New Application 24105 Apiuser17092022-1 <TBA>, apiuser17092022-1@plexe.co.au, 8888888888, Sep 17, 2022",
                    "customerId": "",
                    "partnerId": "",
                    "extraInformation": {
                    "years": "2-5 years", "zipcode": "82772", "industry": "Rice"
                    }
                },
                "customer": {
                    "email": "apiuser17092022-1@plexe.co.au", 
	                "firstName": "Apiuser17092022-1", 
	                "lastName": "<TBA>",
	                "mobileNumber": "8888888888",
	                "password": "DelightWAY12!@"
                }
            }';                   
            
            $request = new Request('POST', $this->BASE_URI.$this->APPLICATION_URI.'create-customer-and-application',$headers, $body);
            
            $res = $client->sendAsync($request)->wait();   
            
            $result = $res->getBody();
            $decoded = json_decode($result);
            return $decoded; 
        }
        catch (RequestException $e)
        {            
            return 'BAD_REQUEST';
        }
    }
    
    public function GetApplicationOffer($key, $applicationId)
    {
            
        try
        {
            $client = new Client();
            $headers = [
                'Accept' => 'application/json',
                'Authorization' => 'Bearer '.$key
            ];
            $request = new Request('GET', $this->BASE_URI.$this->APPLICATION_URI.'application-offer/'.$applicationId.'?force=false', $headers);
            $res = $client->sendAsync($request)->wait();
                
            $result = $res->getBody();
            $decoded = json_decode($result);
            return $decoded;
        }
        catch (RequestException $e)
        {            
            return 'BAD_REQUEST';
        }
    }
        
    public function AddOrUpdateBankingData($key, $applicationId, $bankName)
    {
            
        try
        {
            $client = new Client();
            $headers = [
                'Content-Type' => 'application/json',
                'Accept' => 'application/json',
                'Authorization' => 'Bearer '.$key
            ];
            $body = '{
                  "bankAccounts": [
                    {
                      "bank": "Bank One",
                      "availableBalance": "23333",
                      "accountNumber": "82722892920",
                      "routing": "827282",
                      "accountName": "Bank One Test",
                      "currentBalance": "3455",
                      "accountType": "Current",
                      "transactions": [
                        {
                          "amount": 34,
                          "date": "1962-03-15T21:07:50.047Z",
                          "text": "Transaction One",
                          "isCredit": true,
                          "balance": 345
                        },
                        {
                          "amount": 23,
                          "date": "2020-08-03T08:07:00.597Z",
                          "text": "Transaction Two",
                          "isCredit": true,
                          "balance": 356
                        }
                      ]
                    },
                    {
                      "bank": "Bank Two",
                      "availableBalance": "34553",
                      "accountNumber": "9774789329",
                      "routing": "983738",
                      "accountName": "Bank Two Test",
                      "currentBalance": "8373",
                      "accountType": "Current",
                      "transactions": [
                        {
                          "amount": 34,
                          "date": "1962-03-15T21:07:50.047Z",
                          "text": "Transaction One",
                          "isCredit": true,
                          "balance": 234
                        },
                        {
                          "amount": 67,
                          "date": "2020-08-03T08:07:00.597Z",
                          "text": "Transaction Two",
                          "isCredit": true,
                          "balance": 456
                        }
                      ]
                    }
                  ]
            }';
            $request = new Request('POST', $this->BASE_URI.$this->APPLICATION_URI.'add-or-update-banking-data/'.$applicationId.'?bankName='.$bankName, $headers, $body);
            $res = $client->sendAsync($request)->wait();
            $result = $res->getBody();
            $decoded = json_decode($result);
            return array($res->getStatusCode(), $decoded);
        }
        catch (RequestException $e)
        {            
            return 'BAD_REQUEST';
        }
    }

    public function ProcessApplication($key, $applicationId)
    {
        try
        {
            $client = new Client();
            $headers = [
                'Accept' => 'application/json',
                'Authorization' => 'Bearer '.$key
            ];
            $request = new Request('GET', $this->BASE_URI.$this->APPLICATION_URI.'process-application/'.$applicationId, $headers);
            $res = $client->sendAsync($request)->wait();
            $result = $res->getBody();
            $decoded = json_decode($result);
            return $decoded;
        }
        catch (RequestException $e)
        {            
            return 'BAD_REQUEST';
        }
    }

    public function ApplicationStatus($key, $applicationId)
    {
            
        try
        {
            $client = new Client();
            $headers = [
                'Accept' => 'application/json',
                'Authorization' => 'Bearer '.$key
            ];
            $request = new Request('GET',  $this->BASE_URI.$this->APPLICATION_URI.'application-status/'.$applicationId, $headers);
            $res = $client->sendAsync($request)->wait();
            $result = $res->getBody();
            $decoded = json_decode($result);
            return $decoded;

        }
        catch (RequestException $e)
        {            
            return 'BAD_REQUEST';
        }
    }
}
?>